Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -124,51 +94,116 @@ extension HTTPClient {
+    }
 
-        let begin = index(after: startIndex)
-        let end = index(before: endIndex)
-        return String(self[begin..<end])
+    /// If this collection begins and ends with an ASCII double-quote ("),
+    /// returns a version of self trimmed of those quotes. Otherwise, returns self.
+    fileprivate func trimmingPairedASCIIQuote() -> SubSequence {
+        let quoteChar = UInt8(ascii: "\"")
+        var trimmed = self
+        if trimmed.popFirst() == quoteChar && trimmed.popLast() == quoteChar {
+            return trimmed
+        }
+        return self
+    }
+
+    /// Splits this collection in to a key and value at the first ASCII '=' character.
+    /// Both the key and value are trimmed of ASCII spaces.
+    fileprivate func parseKeyValuePair() -> (key: SubSequence, value: SubSequence)? {
+        guard let keyValueSeparator = self.firstIndex(of: UInt8(ascii: "=")) else {
+            return nil
+        }
+        let trimmedName = self[..<keyValueSeparator].trimmingASCIISpaces()
+        let trimmedValue = self[self.index(after: keyValueSeparator)...].trimmingASCIISpaces()
+        return (trimmedName, trimmedValue)
+    }
+
+    /// Parses this collection as either a key-value pair, or a plain key.
+    /// The returned key is trimmed of ASCII spaces and normalized to lowercase.
+    /// The returned value is trimmed of ASCII spaces.
+    fileprivate func parseCookieComponent() -> (key: String, value: SubSequence?)? {
+        let (trimmedName, trimmedValue) = self.parseKeyValuePair() ?? (self.trimmingASCIISpaces(), nil)
+        guard !trimmedName.isEmpty else {
+            return nil
+        }
+        return (String(decoding: trimmedName, as: UTF8.self).lowercased(), trimmedValue)
     }
 }
 
-extension HTTPClient.Response {
-    /// List of HTTP cookies returned by the server.
-    public var cookies: [HTTPClient.Cookie] {
-        return self.headers["set-cookie"].compactMap { HTTPClient.Cookie(header: $0, defaultDomain: self.host) }
+private let posixLocale: UnsafeMutableRawPointer = {
+    // All POSIX systems must provide a "POSIX" locale, and its date/time formats are US English.
+    // https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap07.html#tag_07_03_05
+    let _posixLocale = newlocale(LC_TIME_MASK | LC_NUMERIC_MASK, "POSIX", nil)!
+    return UnsafeMutableRawPointer(_posixLocale)
+}()
+
+private func parseTimestamp(_ string: String, format: String) -> tm? {
+    var timeComponents = tm()
+    guard swiftahc_cshims_strptime_l(string, format, &timeComponents, posixLocale) else {
+        return nil
+    }
+    return timeComponents
+}
+
+private func parseCookieTime(_ timestampUTF8: String.UTF8View.SubSequence, in header: String) -> Int64? {
+    if timestampUTF8.contains(where: { $0 < 0x20 /* Control characters */ || $0 == 0x7F /* DEL */ }) {
+        return nil
+    }
+    let timestampString: String
+    if timestampUTF8.reversed().starts(with: "TMG".utf8) {
```

Comment:
Yeah, there is no general "hasSuffix" function, so this is the best I could do 😅 

I thought about doing `.reversed().starts(with: "GMT".utf8.reversed())`, but the double-reversing could have a performance/size impact.

Reply:
Honestly I think a one-off helper function will help with clarity here. :wink: